### PR TITLE
youtube songs in csv format

### PR DESCRIPTION
**Pull request by xinye83**
Youtube songs in CSV format: artist name(s), song name, URL
- multiple artist names are separated by &
- different Youtube link formats

### CODE REVIEW

**Comment by YLPeng** [comment]
Looks good!
Do you want to continue this work to try some rust coding?
- parse this csv
- add each song in DB (can use this method https://github.com/xinye83/openKTV/blob/main/KtvService/src/repo/song.rs#L11)
- [optional] add an API endpoint for this process: e.g. user call `/import-songs` with a csv file path, then system will read the file and add songs to the DB

**Comment by xinye83** [comment]
Sounds good.

File: KtvService/src/api/song.rs [hunk 1]
```diff
@@ -43,3 +49,29 @@ pub async fn query_songs(ddb: Data<DBRepository>, query: Json<SongRequest>, para
...
+
+    let mut ids: Vec<u64> = Vec::new();
+
+    for item in reader.deserialize() {
+        let item: SongRequest = match item {
+            Ok(item) => item,
+            Err(err) => return Err(ApiError::CsvReadError(err)),
+        };
+
+        let rtn = ddb.insert_song(item).await;
```
Comment (YLPeng) [inline KtvService/src/api/song.rs:68]: ideally should create a new DB function to insert multiple songs at once, but it's fine for now

**Review by YLPeng — APPROVED** [review]

**Comment by YLPeng** [comment]
LGTM!

**Pull request merged**